smart-contracts/assembly/contracts/FT/WMAS.ts

```diff
@@ -0,0 +1,41 @@
+import { Args, u256ToBytes } from '@massalabs/as-types';
+import { Address, Context, transferCoins } from '@massalabs/massa-as-sdk';
+import { burn } from './burnable/burn';
+import { u256 } from 'as-bignum/assembly/integer/u256';
+import { _mint } from './mintable/mint-internal';
+
+export * from './token';
+
+/**
+ * Wrap wanted value.
+ *
+ * @param _ - unused but mandatory.
+ */
+export function deposit(_: StaticArray<u8>): void {
+  const amount = Context.transferredCoins();
+  const recipient = Context.caller();
+  assert(amount > 0, 'Payment must be more than 0 MAS');
```

Comment:
assert(amount > 0, 'Payment must be more than 0 MAS');
const recipient = Context.caller();